src/hangar/checkout.py

```diff
@@ -73,8 +139,12 @@ def close(self):
         system resources, which may improve performance for repositories with
         multiple simultaneous read checkouts.
         '''
-        for dsetHandle in self.datasets.values():
+        self.__verify_checkout_alive()
+        for dsetHandle in self._datasets.values():
             dsetHandle._close()
+        del self._datasets
+        del self._metadata
+        gc.collect()
```

Comment:
Before this goes in, let's try removing collect after `gc.disable()`, and see if the call to `del` is sufficient to invalidate the weak ref. I'd prefer if we got rid of the call to `collect`

Reply:
So after a bit of testing it turned out this wasn't necessary after all! Nice spot!